velox/exec/WindowPartition.cpp

```diff
@@ -407,26 +407,65 @@ void WindowPartition::updateKRangeFrameBounds(
             mappedFrameColumn,
             flags) == 0) {
       rawFrameBounds[i] = rawPeerBounds[i];
-    } else {
-      // If the search is for a preceding bound then rows between
-      // [0, currentRow] are examined. For following bounds, rows between
-      // [currentRow, numRows()) are checked.
-      if (isPreceding) {
+      continue;
+    }
+
+    auto compareWithLastRow = [&](vector_size_t currentRow) {
+      // For the first row, not to apply the sliding pattern.
+      if (currentRow == startRow) {
+        // For the first row, just returns a arbitrary positive value
+        // to indicate not to use sliding search pattern.
+        return 1;
+      }
+      return data_->compare(
+          partition_[currentRow - 1],
+          partition_[currentRow],
+          mappedFrameColumn,
+          mappedFrameColumn,
+          flags);
+    };
+
+    // If the search is for a preceding bound then rows between
+    // [0, currentRow] are examined. For following bounds, rows between
+    // [currentRow, numRows()) are checked. If the sliding search pattern
+    // is applicable, we also use the last row's search result to determine
+    // the start search bound for current row.
+
+    if (compareWithLastRow(currentRow) == 0) {
+      // As same as last row's searching result.
+      rawFrameBounds[i] = rawFrameBounds[i - 1];
+      continue;
+    }
+
+    if (isPreceding) {
+      // It's the first row or it doesn't match the sliding search pattern.
+      if (compareWithLastRow(currentRow) > 0) {
         start = 0;
-        end = currentRow + 1;
       } else {
+        start = rawFrameBounds[i - 1] == -1
+            ? 0
+            : std::min(rawFrameBounds[i - 1], i);
```

Comment:
Can you elaborate the reason for the use of std::min here ?

Reply:
@aditi-pandit, seems not necessary. Just removed. Thanks!